src/my_os2cond.c

```diff
@@ -67,7 +67,7 @@ int pthread_cond_init(pthread_cond_t *cond, const pthread_condattr_t *attr)
 	cv->semaphore = -1;
 
 	// right now we only handle the clock_id attribute
-	if (attr == NULL || *attr == NULL)
+	if (attr == NULL || *attr == NULL || !verify_memory((ULONG)attr, sizeof(*cond_attr)))

```

Comment:
There is a small bug here: `attr` is a pointer to pointer, we should dereference it to access the second pointer that we need to validate (and then dereference again for `sizeof` to properly catch its type).

```suggestion
	if (attr == NULL || *attr == NULL || !verify_memory((ULONG)*attr, sizeof(**attr)))
```

In the current form this line does nothing because `attr` itself is valid (it's either null or some addr on the stack).